Sources/TuistKit/Services/RunService.swift

```
@@ -0,0 +1,142 @@
+
+        _ = try xcodebuildController.build(.workspace(workspacePath),
+                                           scheme: schemeName,
+                                           clean: false,
+                                           arguments: buildGraphInspector.buildArguments(target: buildableTarget!,
+                                                                                         configuration: nil))
+            .printFormattedOutput()
+            .toBlocking()
+            .last()
+    }
+
+    private func findApp(for target: Target) throws -> AbsolutePath {
+        let appsLocation = FileHandler.shared.locateDirectory("../Library/Developer/Xcode/DerivedData/",
+                                                              traversingFrom: FileHandler.shared.currentPath)?
+            .glob("\(target.name)-*")
+            .first?
+            .glob("Build/Products/Debug-iphonesimulator/*.app")
+
+        guard let locations = appsLocation else { throw RunServiceError.appNotFound(targetName: target.name) }
+        if locations.count == 0 {
+            throw RunServiceError.appNotFound(targetName: target.name)
+        } else {
+            return locations.first!
+        }
+    }
+
+    private func launchApp(for target: Target) throws {
+        let simulator = try findDevice()
+
+        if let simulator = simulator {
+            if simulator.isShutdown {
+                try simulatorController.bootSimulator(simulator)
+            }
+            try installAppBuilt(for: target)
+            try launchAppOnSimulator(for: target)
+        } else {
+            throw RunServiceError.simulatorNotFound
+        }
+    }
+
+    // Sort of wondering if it belongs here or in the SimulatorController class
+    // At some point we can imagine we'll have some arguments to determine which
+    // simulator to pick (device, OS version, etc)
+    private func findDevice() throws -> SimulatorDevice? {
```

I'd move this one into `SimulatorController` and return a `Single<SimulatorDevice>`

```swift
func findAvailableDevice(platform: Platform) -> Single<SimulatorDevice>
```

If a device can't be found we can raise an error instead of returning `nil`.